starlingcaptureapi/fs_watcher.py

```diff
@@ -27,106 +29,163 @@ def caught_and_logged_exceptions(event):
 class FsWatcher:
     """Watches directories for file changes."""
 
-    def __init__(self, organization_id):
-        self.organization_id = organization_id
+    def __init__(self, org_config: dict, collection_id: str = None):
+        self.org_config = org_config
+        self.collection_id = collection_id
+        self.organization_id = org_config.get("id")
+        self.asset_helper = AssetHelper(self.organization_id)
+        self.observer = Observer()
+
+    @staticmethod
+    def start(org_config: dict):
+        FsWatcher(org_config).watch()
+
+    @staticmethod
+    def init_all(
+        all_org_config: config.OrganizationConfig = config.ORGANIZATION_CONFIG,
+    ) -> list[multiprocessing.Process]:
+        """Initialize file watcher processes for the given configuration.
 
-    def watch(self):
-        """
         Args:
-            organization_id: a string identifying the organization we are watching for
+            org_config: configuration for all organizations and their actions
+
+        Returns:
+            list of un-started processes containing FsWatcher instances
         """
-        observer = Observer()
-        patterns = ["*.jpg", "*.jpeg"]
-        asset_helper = AssetHelper(self.organization_id)
-        observer.schedule(
-            self.AddHandler(patterns=patterns).set_org_id(self.organization_id),
-            recursive=True,
-            path=asset_helper.get_assets_add(),
+        procs = []
+        for org_id, org_config in all_org_config.config.items():
+            procs.append(
+                multiprocessing.Process(
+                    name=f"fs_watcher_{org_id}",
+                    target=FsWatcher.start,
+                    args=(org_config,),
```

Comment:
Why do there need to be separate processes for each file watcher? Can one watcher not watch multiple patterns or directories? Or is this just way more efficient in your experience.

Reply:
This is implemented this way as an evolution of where things started off, and because it is simpler implementation-wise to have each watcher be responsible for an organization.

I am not aware of any performance or efficiency considerations having been taken into account here. (Although, as far as I know, so far we only have had one organization sending data at any one time, so it seems like there isn't a need to optimize this.)